crates/glaredb_node/src/session.rs

```diff
@@ -0,0 +1,145 @@
+use glaredb_core::arrays::batch::Batch;
+use glaredb_core::arrays::field::ColumnSchema;
+use glaredb_core::arrays::format::pretty::components::PRETTY_COMPONENTS;
+use glaredb_core::arrays::format::pretty::table::PrettyTable;
+use glaredb_core::engine::single_user::SingleUserEngine;
+use glaredb_error::DbError;
+use glaredb_ext_csv::extension::CsvExtension;
+use glaredb_ext_parquet::extension::ParquetExtension;
+use glaredb_ext_spark::SparkExtension;
+use glaredb_ext_tpch_gen::TpchGenExtension;
+use glaredb_rt_native::runtime::{
+    NativeSystemRuntime,
+    ThreadedNativeExecutor,
+    new_tokio_runtime_for_io,
+};
+use napi_derive::napi;
+
+use crate::errors::Result;
+
+const DEFAULT_TABLE_WIDTH: usize = 100;
+
+pub fn connect() -> napi::Result<NodeSession> {
+    let tokio_rt =
+        new_tokio_runtime_for_io().map_err(|e| napi::Error::from_reason(e.to_string()))?;
+    let runtime = NativeSystemRuntime::new(tokio_rt.handle().clone());
+    let executor =
+        ThreadedNativeExecutor::try_new().map_err(|e| napi::Error::from_reason(e.to_string()))?;
+
+    let engine = SingleUserEngine::try_new(executor, runtime.clone())
+        .map_err(|e| napi::Error::from_reason(e.to_string()))?;
+
+    engine
+        .register_extension(SparkExtension)
+        .map_err(|e| napi::Error::from_reason(e.to_string()))?;
+    engine
+        .register_extension(TpchGenExtension)
+        .map_err(|e| napi::Error::from_reason(e.to_string()))?;
+    engine
+        .register_extension(CsvExtension)
+        .map_err(|e| napi::Error::from_reason(e.to_string()))?;
+    engine
+        .register_extension(ParquetExtension)
+        .map_err(|e| napi::Error::from_reason(e.to_string()))?;
+
+    Ok(NodeSession {
+        tokio_rt,
+        engine: Some(engine),
+    })
+}
+
+#[napi]
+#[derive(Debug)]
+pub struct NodeSession {
+    #[allow(unused)]
+    pub(crate) tokio_rt: tokio::runtime::Runtime,
+    pub(crate) engine: Option<SingleUserEngine<ThreadedNativeExecutor, NativeSystemRuntime>>,
+}
+
+#[napi]
+impl NodeSession {
+    #[napi]
+    pub async unsafe fn sql(&mut self, sql: String) -> napi::Result<NodeQueryResult> {
+        unsafe { self.query(sql).await }
+    }
+
+    #[napi]
+    pub async unsafe fn query(&mut self, sql: String) -> napi::Result<NodeQueryResult> {
```

Comment:
Why are these unsafe?

Reply:
We can probably remove the `mut` here since the single user session has a mutex internally. That should remove the need for unsafe as well.